Specifications/Events.Store/for_CommittedAggregateEventsExtension/given/all_dependencies.cs

```diff
@@ -42,13 +43,18 @@ public class all_dependencies
 
     protected static Contracts.CommittedAggregateEvents with_committed_events(params Contracts.CommittedAggregateEvents.Types.CommittedAggregateEvent[] events)
     {
-        var aggregate_version_after_commit = number_of_events_in_aggregate_when_commit_happened + (ulong)events.Length;
+        var aggregate_version_after_commit = number_of_events_in_aggregate_when_commit_happened;
+        foreach (var e in events)
+        {
+            e.AggregateRootVersion = aggregate_version_after_commit++;
+        }
         var result = new Contracts.CommittedAggregateEvents
         {
             AggregateRootId = aggregate_root_id.ToProtobuf(),
             EventSourceId = event_source_id,
             // The aggregate root version of the committed aggregate events is the version of the last event, meaning the aggregate root version before commit happened + number of events - 1
-            AggregateRootVersion = aggregate_version_after_commit - 1
+            AggregateRootVersion = aggregate_version_after_commit - 1,
```

Comment:
This should also check for the version being 0 before subtracting 1 - right?

Reply:
Yes it should :)